### ISSUE TITLE
Build fails with recent glib version

### ISSUE THREAD

**Issue by rffontenelle** [Buzztrax/buzztrax#91]
Build fails with the following output:

```
  CXX      src/gst/sidsyn/libgstsidsyn_la-sidemu.lo
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/gstreamer-1.0/gst/gst.h:27,
                 from src/gst/dec/bt-dec.h:21,
                 from src/gst/dec/bt-dec.c:62:
src/gst/dec/bt-dec.c:956:14: error: expected '=', ',', ';', 'asm' or '__attribute__' before '-' token
  956 |     buzztrax - dec,
      |              ^
/usr/include/glib-2.0/glib/gmacros.h:735:62: note: in definition of macro 'G_PASTE_ARGS'
  735 | #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
      |                                                              ^~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:257:41: note: in expansion of macro 'G_PASTE'
  257 | GST_PLUGIN_EXPORT const GstPluginDesc * G_PASTE(gst_plugin_, G_PASTE(name, _get_desc)) (void); \
      |                                         ^~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:736:47: note: in expansion of macro 'G_PASTE_ARGS'
  736 | #define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1, identifier2)
      |                                               ^~~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:257:62: note: in expansion of macro 'G_PASTE'
  257 | GST_PLUGIN_EXPORT const GstPluginDesc * G_PASTE(gst_plugin_, G_PASTE(name, _get_desc)) (void); \
      |                                                              ^~~~~~~
src/gst/dec/bt-dec.c:954:1: note: in expansion of macro 'GST_PLUGIN_DEFINE'
  954 | GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
      | ^~~~~~~~~~~~~~~~~
src/gst/dec/bt-dec.c:956:14: error: expected '=', ',', ';', 'asm' or '__attribute__' before '-' token
  956 |     buzztrax - dec,
      |              ^
/usr/include/glib-2.0/glib/gmacros.h:735:62: note: in definition of macro 'G_PASTE_ARGS'
  735 | #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
      |                                                              ^~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:258:24: note: in expansion of macro 'G_PASTE'
  258 | GST_PLUGIN_EXPORT void G_PASTE(gst_plugin_, G_PASTE(name, _register)) (void); \
      |                        ^~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:736:47: note: in expansion of macro 'G_PASTE_ARGS'
  736 | #define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1, identifier2)
      |                                               ^~~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:258:45: note: in expansion of macro 'G_PASTE'
  258 | GST_PLUGIN_EXPORT void G_PASTE(gst_plugin_, G_PASTE(name, _register)) (void); \
      |                                             ^~~~~~~
src/gst/dec/bt-dec.c:954:1: note: in expansion of macro 'GST_PLUGIN_DEFINE'
  954 | GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
      | ^~~~~~~~~~~~~~~~~
src/gst/dec/bt-dec.c:956:14: error: expected '=', ',', ';', 'asm' or '__attribute__' before '-' token
  956 |     buzztrax - dec,
      |              ^
/usr/include/glib-2.0/glib/gmacros.h:735:62: note: in definition of macro 'G_PASTE_ARGS'
  735 | #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
      |                                                              ^~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:276:1: note: in expansion of macro 'G_PASTE'
  276 | G_PASTE(gst_plugin_, G_PASTE(name, _get_desc)) (void) \
      | ^~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:736:47: note: in expansion of macro 'G_PASTE_ARGS'
  736 | #define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1, identifier2)
      |                                               ^~~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:276:22: note: in expansion of macro 'G_PASTE'
  276 | G_PASTE(gst_plugin_, G_PASTE(name, _get_desc)) (void) \
      |                      ^~~~~~~
src/gst/dec/bt-dec.c:954:1: note: in expansion of macro 'GST_PLUGIN_DEFINE'
  954 | GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
      | ^~~~~~~~~~~~~~~~~
src/gst/dec/bt-dec.c:956:14: error: expected '=', ',', ';', 'asm' or '__attribute__' before '-' token
  956 |     buzztrax - dec,
      |              ^
/usr/include/glib-2.0/glib/gmacros.h:735:62: note: in definition of macro 'G_PASTE_ARGS'
  735 | #define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
      |                                                              ^~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:282:1: note: in expansion of macro 'G_PASTE'
  282 | G_PASTE(gst_plugin_, G_PASTE(name, _register)) (void) \
      | ^~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:736:47: note: in expansion of macro 'G_PASTE_ARGS'
  736 | #define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1, identifier2)
      |                                               ^~~~~~~~~~~~
/usr/include/gstreamer-1.0/gst/gstplugin.h:282:22: note: in expansion of macro 'G_PASTE'
  282 | G_PASTE(gst_plugin_, G_PASTE(name, _register)) (void) \
      |                      ^~~~~~~
src/gst/dec/bt-dec.c:954:1: note: in expansion of macro 'GST_PLUGIN_DEFINE'
  954 | GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
      | ^~~~~~~~~~~~~~~~~
  CXX      src/gst/sidsyn/libgstsidsyn_la-voice.lo
make[2]: *** [Makefile:6171: src/gst/dec/libbuzztraxdec_la-bt-dec.lo] Error 1
```

**Full build log:**
[build.log](https://github.com/Buzztrax/buzztrax/files/4203415/build.log)

**Environment:**
* Distro: Arch Linux 64-bit
* Packages:
   * [buzztrax](https://aur.archlinux.org/packages/buzztrax) 0.10.2
   * [glib2](https://www.archlinux.org/packages/core/x86_64/glib2/) 2.62.4
   * [gstreamer](https://www.archlinux.org/packages/extra/x86_64/gstreamer/) 1.16.2

**Comment by trusktr** [comment]
By any chance is this a newer version of a build tool causing the error? The reason I thought this is because the version of Buzztrax being built is 0.10.2, downloaded from http://files.buzztrax.org/releases/buzztrax-0.10.2.tar.gz. More details in the Arch build file, https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD, which was last updated in 2017. I assume it probably worked at that time.

**Comment by rffontenelle** [comment]
@trusktr I looks to me as that GStreamer or Glib2 libraries have changed in a way that they are incompatible with buzztrax, as I see no GCC error in this trace,..

**Comment by trusktr** [comment]
Ah ok. I assumed a syntax error was from a change in tooling. Could a change in gstreamer or glib2 cause a syntax error?

**Comment by rffontenelle** [comment]
It is always possible that a new version of a library breaks compatibility with a previous one. Since Arch is rolling release, it happens sometimes.

**Comment by trusktr** [comment]
Stefan said he may take a look! https://gitter.im/Buzztrax/buzztrax

**Comment by wadem** [comment]
hitting build issues as well on OSX Mojave too;

`src/gst/dec/bt-dec.c:956:14: error: expected ';' after top level declarator
    buzztrax - dec,
             ^
src/gst/dec/bt-dec.c:954:1: error: variable has incomplete type 'void'
GST_PLUGIN_DEFINE (GST_VERSION_MAJOR,
^
/usr/local/Cellar/gstreamer/1.16.2/include/gstreamer-1.0/gst/gstplugin.h:258:24: note:
      expanded from macro 'GST_PLUGIN_DEFINE'
GST_PLUGIN_EXPORT void G_PASTE(gst_plugin_, G_PASTE(name, _register)) (void); \
                       ^
/usr/local/Cellar/glib/2.64.3/include/glib-2.0/glib/gmacros.h:736:47: note:
      expanded from macro 'G_PASTE'
#define G_PASTE(identifier1,identifier2)      G_PASTE_ARGS (identifier1...
                                              ^
/usr/local/Cellar/glib/2.64.3/include/glib-2.0/glib/gmacros.h:735:47: note:
      expanded from macro 'G_PASTE_ARGS'
#define G_PASTE_ARGS(identifier1,identifier2) identifier1 ## identifier2
                                              ^
<scratch space>:74:1: note: expanded from here
gst_plugin_buzztrax
^
src/gst/dec/bt-dec.c:956:14: error: expected ';' after top level declarator
    buzztrax - dec,
             ^
src/gst/dec/bt-dec.c:956:14: error: expected ';' after top level declarator
4 errors generated.
make[2]: *** [src/gst/dec/libbuzztraxdec_la-bt-dec.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2`

**Comment by rffontenelle** [comment]
Fixed in a89abcf781c8a6b6940220fd291858c9d5b1c89c
